src/fmu/dataio/_export_item.py

```diff
@@ -318,11 +366,66 @@ def _data_process_object(self):
 
         if self.subtype == "RegularSurface":
             self._data_process_object_regularsurface()
+        elif self.subtype == "Grid":
+            self._data_process_object_grid()
+        elif self.subtype == "GridProperty":
+            self._data_process_object_gridproperty()
         elif self.subtype == "Polygons":
             self._data_process_object_polygons()
         elif self.subtype == "DataFrame":
             self._data_process_object_dataframe()
 
+    def _data_process_object_grid(self):
```

Comment:
(Ref comment above; this would then be `_data_process_object_cornerpointgrid` or something similar.)